File: javascript/hamming/hamming.js
var Hamming = function () {};

Hamming.prototype.compute = function (s1, s2) {
    'use strict';

    var len1 = s1.length;
    var len2 = s2.length;

    if (len1 !== len2) {
        throw new Error('DNA strands must be of equal length.');
    }

    var diffs = 0;
    for (var i = 0; i < len1; i++) {
        if (s1[i] !== s2[i]) {
            diffs ++;
        }
    }

    return diffs;
};

module.exports = Hamming;

File: go/hamming/hamming.go
package hamming

import (
	"errors"
	"strings"
)

// calculate distance between a and b
func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return 0, errors.New("length not equal")
	}
	a = strings.ToUpper(a)
	b = strings.ToUpper(b)
	ret := 0
	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			ret++
		}
	}
	return ret, nil
}
